apps/veil/src/entities/leaderboard/ui/table.tsx

```diff
@@ -1,76 +1,84 @@
+const Tabs = {
+  AllLPs: 'All LPs',
+  MyLPs: 'My LPs',
+} as const;
+
+type Tab = (typeof Tabs)[keyof typeof Tabs];
+
+export const LeaderboardTable = observer(() => {
+  const { connected, subaccount } = connectionStore;
+  const totalRef = useRef<number>(0);
+  const params = useParams<{ epoch: string }>();
+  const epoch = Number(params?.epoch);
   const searchParams = useSearchParams();
   const page = Number(searchParams?.get('page') ?? 1);
   const [currentPage, setCurrentPage] = useState(page);
   const [parent] = useAutoAnimate();
-  const [quote, setQuote] = useState<AssetSelectorValue>();
-  const [tab, setTab] = useState<'All LPs' | 'My LPs'>('All LPs');
+  const [tab, setTab] = useState<Tab>(Tabs.AllLPs);
   const [limit, setLimit] = useState(10);
-  const quoteAssetId = getAssetId(quote);
-  const { getTableHeader, sortBy } = useSortableTableHeaders();
+  const { getTableHeader, sortBy } = useSortableTableHeaders<LpLeaderboardSortKey>('points');
 
-  const { data: summary } = useTournamentSummary({
-    limit: 1,
-    page: 1,
-  });
+  const getAssetMetadata = useGetMetadata();
+  const { data: umMetadata } = useStakingTokenMetadata();
 
   const {
     data: leaderboard,
-    error,
-    isLoading,
-  } = useLeaderboard({
+    error: leaderboardError,
+    isPending: leaderboardLoading,
+  } = useLpLeaderboard({
+    epoch,
+    page: currentPage,
     limit,
-    offset: (currentPage - 1) * limit,
-    quote: quoteAssetId,
-    startBlock: summary?.[0]?.start_block,
-    endBlock: summary?.[0]?.end_block,
+    sortKey: sortBy.key,
+    sortDirection: sortBy.direction,
+    isActive: tab === Tabs.AllLPs,
   });
 
-  const { data: assets } = useAssets();
-  const { data: balances } = useBalances();
-  const { data: positions, totalCount } = leaderboard ?? {};
-  totalCountRef.current = totalCount ?? totalCountRef.current;
+  const isMyTab = tab === Tabs.MyLPs;
+  const {
+    data: myLeaderboard,
+    error: myLeaderboardError,
+    isPending: myLeaderboardLoading,
+  } = useMyLpLeaderboard({
+    subaccount,
+    epoch,
+    page: currentPage,
+    limit,
+    sortKey: sortBy.key,
+    sortDirection: sortBy.direction,
+    isActive: isMyTab,
+  });
 
-  const sortedPositions = useMemo(() => {
-    return orderBy(
-      (positions ?? []).map(position => ({
-        ...position,
-        sortValues: {
-          executions: position.executions,
-          pnlPercentage: position.pnlPercentage,
-          points: Math.abs(position.pnlPercentage),
-          age: (position.closingTime ?? 0) - position.openingTime,
-        },
-      })),
-      `sortValues.${sortBy.key}`,
-      sortBy.direction,
-    );
-  }, [positions, sortBy]);
+  const [positions, total, error, isLoading] = isMyTab
+    ? [myLeaderboard?.data, myLeaderboard?.total, myLeaderboardError, myLeaderboardLoading]
+    : [leaderboard?.data, leaderboard?.total, leaderboardError, leaderboardLoading];
+  totalRef.current = total ?? totalRef.current;
 
   if (error) {
```

Comment:
suggestion: wrap error content into Card too. For now the error hangs within the page and it's not easy to understand what it relates to